src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const split = fromFormat[3];
+  const dayOfYear = date.split(split);
+  const tab = [];
+
+  for (let i = 0; i < 3; i++) {
+    for (let j = 0; j < 3; j++) {
+      if (fromFormat[i] === toFormat[j]) {
+        tab[j] = dayOfYear[i];
+      }
+    }
+  }
+
+  let year;
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dayOfYear[i];
+    }
+  }
+
+  if (year.length === 2 && year < 30) {
+    year = '20' + year;
+  }
+
+  if (year.length === 2 && year >= 30) {
+    year = '19' + year;
+  }
+
+  for (let i = 0; i < 3; i++) {
+    if (toFormat[i] === 'YY' && year.length === 4) {
+      year = year.slice(2, 4);
+    }
+  }
```

Comment:
The loop here is unnecessary since you are only interested in modifying the year if the 'toFormat' requires 'YY'. You can check directly for 'YY' in 'toFormat' without looping through all elements.